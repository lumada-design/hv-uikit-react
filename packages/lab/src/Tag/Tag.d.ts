import * as React from "react";
import { StandardProps } from "@mui/material";

export type Semantic =
  | "sema1"
  | "sema2"
  | "sema3"
  | "sema4"
  | "sema5"
  | "sema6"
  | "sema7"
  | "sema8"
  | "sema9"
  | "sema10"
  | "sema11"
  | "sema12"
  | "sema13"
  | "sema14"
  | "sema15"
  | "sema16"
  | "sema17"
  | "sema18"
  | "sema19"
  | "sema20";

  export type HvTagClassKey =
  | "label"
  | "square"
  | "round"
  | "deleteIconSmall"
  | "disabled";



  export interface HvTagsProps extends StandardProps<React.HTMLAttributes<HTMLDivElement>, HvTagClassKey, "color"> {
  /**
   * An array of strings representing the colors to override in the icon.
   * Each element inside the array will override a diferent color.
   */
   color?: string | string[];

  /**
   * Custom text to show in place of count.
   * Note showCount and label are mutually exclusive.
   * showCount is ignored when label is specified at the same time.
   */
   label?: string;
  /**
   * Sets one of the standard semantic palette colors of the icon
   */
  semantic?: Semantic;


  shape: string,

  /**
 * The disabled property of the Tag.
 */
   disabled: string,

/**
 * The cancel icon is shown based on this property
 */
 showCancelIcon:boolean,

}

export default function HvTag(props: HvTagsProps): JSX.Element | null;
