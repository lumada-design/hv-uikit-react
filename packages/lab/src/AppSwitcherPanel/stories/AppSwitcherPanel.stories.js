/* eslint-disable no-alert */
import React from "react";

import { Tool, PingPong, GameController, Champion } from "@hv/uikit-react-icons";

import HvAppSwitcherPanel from "..";

export default {
  title: "Lab/AppSwitcherPanel",
  parameters: {
    componentSubtitle: null,
    usage: 'import { HvAppSwitcherPanel } from "@hv/uikit-react-lab";',
  },
  component: HvAppSwitcherPanel,
};

export const Main = () => {
  const boxStyles = {
    width: 32,
    height: 32,
  };

  const handlesActionSelectedCallback = (application) => {
    return window.location.href.startsWith(application.url);
  };

  const applicationsList = [
    {
      iconUrl: "",
      description: "Application without a name should not appear",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App without url",
      iconUrl: "",
      description: "Application without a url should not appear",
      target: "_top",
    },
    {
      name: "",
      iconUrl: "",
      description: "Application with an empty name should not appear",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App with empty url",
      iconUrl: "",
      url: "",
      description: "Application with an empty url should not appear",
      target: "_top",
    },
    {
      name: "UI-KIT Storybook",
      iconUrl: "https://pbs.twimg.com/profile_images/1100804485616566273/sOct-Txm_400x400.png",
      description: "This is the Storybook for the UI-KIT project",
      url: "https://lumada-design.github.io/uikit/v2.x/",
      target: "_top",
    },
    {
      name: "UI-KIT GitHub (New Tab)",
      iconElement: <Tool boxStyles={boxStyles} />,
      description: "This is the UI-KIT repository on Github",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_blank",
    },
    {
      name: "App with a bigger name than the other just to showcase the truncation on the AppSwitcherPanel",
      iconElement: <Tool boxStyles={boxStyles} />,
      description: "App 1 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "No Icon App",
      description: "This is an App without an icon, URL is set to the UI-KIT storybook",
      url: "https://github.com/lumada-design/hv-uikit-react",
    },
    {
      name: "No Description App",
      url: "https://github.com/lumada-design/hv-uikit-react",
    },
  ];

  return (
    <div style={{ height: "400px" }}>
      <HvAppSwitcherPanel
        applications={applicationsList}
        isActionSelectedCallback={handlesActionSelectedCallback}
        isOpen
      />
    </div>
  );
};

// Sample 2 - Simple example with footer
export const sample2 = () => {
  const boxStyles = {
    width: 32,
    height: 32,
  };

  const handlesActionSelectedCallback = (application) => {
    return window.location.href.startsWith(application.url);
  };

  const applicationsList = [
    {
      name: "App with a bigger name than the others just to showcase the truncation on the AppSwitcherPanel",
      iconUrl: "https://pbs.twimg.com/profile_images/1100804485616566273/sOct-Txm_400x400.png",
      description: "This is the Storybook for the UI-KIT project",
      url: "https://lumada-design.github.io/uikit/v2.x/",
      target: "_top",
    },
    {
      name: "UI-KIT GitHub (New Tab)",
      iconElement: <Tool boxStyles={boxStyles} />,
      description: "This is the UI-KIT repository on Github",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_blank",
    },
    {
      name: "No Icon App",
      description: "This is an App without an icon, URL is set to the UI-KIT storybook",
      url: "https://github.com/lumada-design/hv-uikit-react",
    },
    {
      name: "No Description App",
      url: "https://github.com/lumada-design/hv-uikit-react",
    },
  ];

  return (
    <div style={{ height: "400px" }}>
      <HvAppSwitcherPanel
        title="Dummy Apps with footer and a really big name to not fit in the container"
        applications={applicationsList}
        isActionSelectedCallback={handlesActionSelectedCallback}
        footer={<div>This is the footer</div>}
        isOpen
      />
    </div>
  );
};

sample2.story = {
  parameters: {
    docs: {
      storyDescription: "Simple example with footer",
    },
  },
};

// Sample 3 - Simple example with a big list of applications
export const sample3 = () => {
  const handlesActionSelectedCallback = (application) => {
    return window.location.href.startsWith(application.url);
  };

  const getDummyApplicationsList = () => {
    const dummyApplicationsList = [];

    for (let index = 1; index <= 100; index += 1) {
      dummyApplicationsList.push({
        name:
          index % 3 === 0
            ? `Application ${index} is an application with a big name`
            : `Application ${index}`,
        iconUrl: `https://i.picsum.photos/id/${index}/32/32.jpg`,
        description: `This is the auto-generated application number ${index}. Note: All the apps redirect to the UI-KIT storybook`,
        url: "https://github.com/lumada-design/hv-uikit-react",
        target: index % 2 === 0 ? "_top" : "_blank",
      });
    }
    return dummyApplicationsList;
  };

  return (
    <div style={{ height: "400px" }}>
      <HvAppSwitcherPanel
        title="Big list of applications"
        applications={getDummyApplicationsList()}
        isActionSelectedCallback={handlesActionSelectedCallback}
        footer={<div>This is the footer</div>}
        isOpen
      />
    </div>
  );
};

sample3.story = {
  parameters: {
    docs: {
      storyDescription: "Simple example with a big list of applications",
    },
  },
};

// Sample 4 - Example with a custom header
export const sample4 = () => {
  const boxStyles = {
    width: 32,
    height: 32,
  };

  const handlesActionSelectedCallback = (application) => {
    return window.location.href.startsWith(application.url);
  };

  const applicationsList = [
    {
      name: "App with a bigger name than the others just to showcase the truncation on the AppSwitcherPanel",
      iconUrl: "https://pbs.twimg.com/profile_images/1100804485616566273/sOct-Txm_400x400.png",
      description: "This is the Storybook for the UI-KIT project",
      url: "https://lumada-design.github.io/uikit/v2.x/",
      target: "_top",
    },
    {
      name: "UI-KIT GitHub (New Tab)",
      iconElement: <Tool boxStyles={boxStyles} />,
      description: "This is the UI-KIT repository on Github",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_blank",
    },
    {
      name: "No Icon App",
      description: "This is an App without an icon, URL is set to the UI-KIT storybook",
      url: "https://github.com/lumada-design/hv-uikit-react",
    },
    {
      name: "No Description App",
      url: "https://github.com/lumada-design/hv-uikit-react",
    },
  ];

  return (
    <div style={{ height: "400px" }}>
      <HvAppSwitcherPanel
        title="Custom header"
        applications={applicationsList}
        isActionSelectedCallback={handlesActionSelectedCallback}
        header={<div style={{ backgroundColor: "lightgreen" }}>This is the custom header</div>}
        isOpen
      />
    </div>
  );
};

sample4.story = {
  parameters: {
    docs: {
      storyDescription: "Example with a custom header",
    },
  },
};

// Sample 5 - Using icon components mixed with url icons
export const sample5 = () => {
  const boxStyles = {
    width: 32,
    height: 32,
  };

  const handlesActionSelectedCallback = (application) => {
    return window.location.href.startsWith(application.url);
  };

  const applicationsList = [
    {
      name: "App 1 - Icon Tool",
      iconElement: <Tool boxStyles={boxStyles} />,
      description: "App 1 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App2 - Url icon",
      iconUrl: "https://pbs.twimg.com/profile_images/1100804485616566273/sOct-Txm_400x400.png",
      description: "App 2 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 3 - Icon PingPong",
      iconElement: <PingPong boxStyles={boxStyles} />,
      description: "App 3 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 4 - No icon",
      description: "App 4 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 5 - Icon GameController",
      iconElement: <GameController boxStyles={boxStyles} />,
      description: "App 5 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App6 - Wrong url icon",
      iconUrl: "http://invalidurl.noicon",
      description: "App 6 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 7 - Icon Champion",
      iconElement: <Champion boxStyles={boxStyles} />,
      description: "App 7 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "ýö ! ' º~ %&#%ç e 合気道",
      iconElement: <Champion boxStyles={boxStyles} />,
      description: "App 8 Special characters",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
  ];

  return (
    <div style={{ height: "500px" }}>
      <HvAppSwitcherPanel
        applications={applicationsList}
        isActionSelectedCallback={handlesActionSelectedCallback}
        isOpen
      />
    </div>
  );
};

sample5.story = {
  parameters: {
    docs: {
      storyDescription: "Using icon components mixed with url icons",
    },
  },
};

// Sample 6 - Alerts on which menu was clicked
export const sample6 = () => {
  const boxStyles = {
    width: 32,
    height: 32,
  };

  const handleActionClicked = (event, application) => {
    event.preventDefault();

    alert(`The clicked application was: ${application.name}`);
  };

  const applicationsList = [
    {
      name: "App 1 - Icon Tool",
      iconElement: <Tool boxStyles={boxStyles} />,
      description: "App 1 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App2 - Url icon",
      iconUrl: "https://pbs.twimg.com/profile_images/1100804485616566273/sOct-Txm_400x400.png",
      description: "App 2 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 3 - Icon PingPong",
      iconElement: <PingPong boxStyles={boxStyles} />,
      description: "App 3 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 4 - No icon",
      description: "App 4 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 5 - Icon GameController",
      iconElement: <GameController boxStyles={boxStyles} />,
      description: "App 5 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App6 - Wrong url icon",
      iconUrl: "http://invalidurl.noicon",
      description: "App 6 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "App 7 - Icon Champion",
      iconElement: <Champion boxStyles={boxStyles} />,
      description: "App 7 description",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
    {
      name: "ýö ! ' º~ %&#%ç e 合気道",
      iconElement: <Champion boxStyles={boxStyles} />,
      description: "App 8 Special characters",
      url: "https://github.com/lumada-design/hv-uikit-react",
      target: "_top",
    },
  ];

  return (
    <div style={{ height: "500px" }}>
      <HvAppSwitcherPanel
        applications={applicationsList}
        onActionClickedCallback={handleActionClicked}
        isOpen
      />
    </div>
  );
};

sample6.story = {
  parameters: {
    docs: {
      storyDescription: "Alerts on which menu was clicked",
    },
  },
};
