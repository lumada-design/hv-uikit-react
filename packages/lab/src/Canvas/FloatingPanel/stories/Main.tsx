import { css } from "@emotion/css";
import {
  HvCanvasFloatingPanel,
  HvCanvasFloatingPanelProps,
} from "@hitachivantara/uikit-react-lab";

const tabs = [
  {
    id: 0,
    title: "Tab 1",
  },
  {
    id: 1,
    title: "Tab 2",
  },
];

const classes = { root: css({ position: "relative" }) }; // for Storybook purposes

export const MainStory = (props: HvCanvasFloatingPanelProps) => (
  <HvCanvasFloatingPanel {...props} className={classes.root} tabs={tabs}>
    Content
  </HvCanvasFloatingPanel>
);
