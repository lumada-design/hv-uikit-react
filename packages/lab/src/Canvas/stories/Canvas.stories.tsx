import { useEffect, useState } from "react";
import { css } from "@emotion/css";
import { Meta, StoryObj } from "@storybook/react";
import { Background, BackgroundVariant, MiniMap } from "reactflow";
import {
  HvButton,
  HvDropDownMenu,
  HvIconButton,
  HvInlineEditor,
  HvLoading,
  theme,
} from "@hitachivantara/uikit-react-core";
import {
  Backwards,
  Calendar,
  DataSource,
  Plane,
  Pod,
  Redo,
  Schema,
  Undo,
  User,
} from "@hitachivantara/uikit-react-icons";
import {
  HvCanvasPanel,
  HvCanvasTab,
  HvCanvasToolbar,
  HvFlow,
} from "@hitachivantara/uikit-react-lab";

import { ListView } from "./ListView/ListView";
import { eds, nds, nodeTypes } from "./nodes";
import { TreeView } from "./TreeView";

const meta: Meta = {
  title: "Lab/Canvas",
};
export default meta;

export const Main: StoryObj = {
  args: {
    title: "Semantic Model Name",
  },
  argTypes: {
    classes: { control: { disable: true } },
  },
  render: () => {
    const [tabs, setTabs] = useState<HvCanvasTab[]>([]);
    const [selectedTab, setSelectedTab] = useState<string>("1");

    const classes = {
      separator: css({
        height: 30,
        width: 1,
        backgroundColor: theme.colors.atmo4,
        margin: `0 ${theme.space.xs}`,
      }),
    };
    const backButton = (
      <HvButton aria-label="Back" icon>
        <Backwards />
      </HvButton>
    );

    const title = (
      <HvInlineEditor value="Semantic Model Name" variant="title4" />
    );

    const Separator = () => <div className={classes.separator} />;

    const renderIcon = (Icon: React.ElementType) => () => <Icon />;

    useEffect(() => {
      setTabs([
        {
          id: "1",
          content: (
            <>
              <DataSource />
              Add Data
            </>
          ),
        },
        {
          id: "2",
          content: (
            <>
              <Schema />
              Model Structure
            </>
          ),
        },
      ]);
    }, []);

    return tabs && tabs.length > 0 ? (
      <div
        style={{
          width: "100%",
          height: "100vh",
          position: "relative",
        }}
      >
        <HvFlow
          style={{ width: "100%", height: "100vh" }}
          snapGrid={[1, 1]}
          snapToGrid
          nodes={nds}
          edges={eds}
          nodeTypes={nodeTypes}
        >
          <Background
            color={theme.palette.slate[950]}
            variant={BackgroundVariant.Dots}
            gap={16}
          />
          <MiniMap />
        </HvFlow>
        <HvCanvasPanel
          tabs={tabs}
          onTabChange={(tabId) => setSelectedTab(tabId)}
        >
          {selectedTab === "1" && <TreeView />}
          {selectedTab === "2" && <ListView />}
        </HvCanvasPanel>
        <HvCanvasToolbar backButton={backButton} title={title}>
          <HvIconButton title="Undo">
            <Undo />
          </HvIconButton>
          <HvIconButton title="Redo">
            <Redo />
          </HvIconButton>
          <Separator />
          <HvIconButton title="Add">
            <Pod />
          </HvIconButton>
          <Separator />
          <HvButton variant="primary">Save</HvButton>
          <HvButton variant="primaryGhost">Cancel</HvButton>
          <Separator />
          <HvDropDownMenu
            placement="right"
            onClick={(e, item) => console.log(item.label)}
            dataList={[
              { label: "Label 1", icon: renderIcon(User) },
              { label: "Label 2", icon: renderIcon(Calendar), disabled: true },
              { label: "Label 3", icon: renderIcon(Plane) },
            ]}
          />
        </HvCanvasToolbar>
      </div>
    ) : (
      <HvLoading />
    );
  },
};
