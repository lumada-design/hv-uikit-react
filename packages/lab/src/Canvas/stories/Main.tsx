import { useState } from "react";
import { css, cx } from "@emotion/css";
import { BackgroundVariant } from "reactflow";
import {
  HvButton,
  HvDropDownMenu,
  HvIconButton,
  HvInlineEditor,
  theme,
} from "@hitachivantara/uikit-react-core";
import {
  Backwards,
  Calendar,
  DataSource,
  Plane,
  Redo,
  Schema,
  Undo,
  User,
} from "@hitachivantara/uikit-react-icons";
import {
  HvCanvasPanel,
  HvCanvasToolbar,
  HvFlow,
  HvFlowBackground,
  HvFlowMinimap,
} from "@hitachivantara/uikit-react-lab";

// The code is available here: https://github.com/lumada-design/hv-uikit-react/tree/master/packages/lab/src/Canvas/stories
import { ListView } from "./ListView/ListView";
import { eds, nds, nodeTypes } from "./nodes";
import { TreeView } from "./TreeView";

const classes = {
  separator: css({
    height: 30,
    width: 1,
    backgroundColor: theme.colors.atmo4,
    margin: `0 ${theme.space.xs}`,
  }),
  flow: css({ width: "100%", height: "100vh" }),
  toolbarFull: css({
    left: 0,
    right: 0,
    marginLeft: "auto",
    marginRight: "auto",
    width: `calc(100% - 2 * ${theme.space.md})`,
  }),
  toolbarMin: css({
    right: theme.space.md,
    width: `calc(100% - 320px - 2 * ${theme.space.md})`,
  }),
};

const tabs = [
  {
    id: "1",
    content: (
      <>
        <DataSource />
        Add Data
      </>
    ),
  },
  {
    id: "2",
    content: (
      <>
        <Schema />
        Model Structure
      </>
    ),
  },
];

const Separator = () => <div className={classes.separator} />;

export const MainStory = () => {
  const [selectedTab, setSelectedTab] = useState<string>("1");
  const [opened, setOpened] = useState(false);

  return (
    <>
      <HvFlow
        className={classes.flow}
        snapGrid={[1, 1]}
        snapToGrid
        nodes={nds}
        edges={eds}
        nodeTypes={nodeTypes}
      >
        <HvFlowBackground
          color={theme.colors.secondary_80}
          variant={BackgroundVariant.Dots}
          gap={16}
          style={{ color: "red" }}
        />
        <HvFlowMinimap />
      </HvFlow>
      <HvCanvasToolbar
        className={cx({
          [classes.toolbarFull]: !opened,
          [classes.toolbarMin]: opened,
        })}
        backButton={
          <HvButton aria-label="Back" icon>
            <Backwards />
          </HvButton>
        }
        title={<HvInlineEditor value="Semantic Model Name" variant="title4" />}
      >
        <HvIconButton title="Undo">
          <Undo />
        </HvIconButton>
        <HvIconButton title="Redo">
          <Redo />
        </HvIconButton>
        <Separator />
        <HvButton variant="primary">Save</HvButton>
        <HvButton variant="primaryGhost">Cancel</HvButton>
        <Separator />
        <HvDropDownMenu
          placement="right"
          onClick={(e, item) => console.log(item.label)}
          dataList={[
            { label: "Label 1", icon: <User /> },
            { label: "Label 2", icon: <Calendar />, disabled: true },
            { label: "Label 3", icon: <Plane /> },
          ]}
        />
      </HvCanvasToolbar>
      <HvCanvasPanel
        open={opened}
        tabs={tabs}
        onTabChange={(event, tabId) => setSelectedTab(tabId)}
        onToggle={(event, value) => setOpened(value)}
      >
        {selectedTab === "1" && <TreeView />}
        {selectedTab === "2" && <ListView />}
      </HvCanvasPanel>
    </>
  );
};
