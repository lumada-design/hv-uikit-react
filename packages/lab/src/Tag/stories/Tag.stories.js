import * as React from "react";

import { HvTag } from "../Tag";

export default {
  title: "Lab/Tag",
  parameters: {
    componentSubtitle: "This component is deprecated. Please use the one in Core.",
    usage: 'import { HvTag } from "@hitachivantara/uikit-react-core"',

    dsVersion: "3.3.0",
  },
  component: HvTag,
};

export const Default = () => {
  return (
    <>
      <HvTag category="informational" id="Tag1">
        Hello
      </HvTag>
      <HvTag category="informational" id="Tag2" showcancelicon>
        How
      </HvTag>
      <HvTag category="informational" id="Tag3" shape="round">
        Are
      </HvTag>
      <HvTag category="informational" id="Tag4" showcancelicon shape="round">
        You
      </HvTag>
    </>
  );
};
Default.decorators = [
  (Story) => (
    <div
      style={{
        display: "inline",
        alignItems: "center",
        width: 400,
        justifyContent: "space-between",
      }}
    >
      <Story />
    </div>
  ),
];

export const Disabled = () => {
  return (
    <>
      <HvTag variant="informational" id="Tag1" disabled>
        Hello
      </HvTag>
      <HvTag variant="informational" id="Tag2" disabled showcancelicon>
        Hello
      </HvTag>
      <HvTag variant="informational" id="Tag2" disabled>
        Hello
      </HvTag>
      <HvTag variant="informational" id="Tag3" disabled shape="round">
        Example
      </HvTag>
    </>
  );
};
Disabled.decorators = [
  (Story) => (
    <div
      style={{
        display: "inline-block",
        alignItems: "center",
        width: 400,
        justifyContent: "space-between",
      }}
    >
      <Story />
    </div>
  ),
];

export const Variant = () => {
  return (
    <>
      <HvTag semantic="sema1">Hello</HvTag>
      <HvTag semantic="sema2">Hello</HvTag>
      <HvTag semantic="sema3">Hello</HvTag>
      <HvTag semantic="sema4">Hello</HvTag>
    </>
  );
};
Variant.decorators = [
  (Story) => (
    <div
      style={{
        display: "inline-block",
        alignItems: "center",
        width: 400,
        justifyContent: "space-between",
      }}
    >
      <Story />
    </div>
  ),
];
