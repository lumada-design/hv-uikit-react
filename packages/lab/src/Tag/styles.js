import { outlineStyles } from "../../../core/src/Focus/styles";

const styles = (theme) => ({
  root: {
    position: "relative",
    cursor: "pointer",
    display: "flex",
    justifyContent: "center",
    flexWrap: "wrap",
    backgroundColor: theme.hv.palette.semantic.sema7,
    margin: "0 8px 8px 0",
    float: "left",
    minWidth: 70,
    padding: theme.hvSpacing(0, "xs"),
    height: 20,
    minHeight: 20,
    textAlign: "center",
    " &:hover": {
      "&:not(disabled)": {
        "& > div svg": {
          backgroundColor: theme.hv.palette.atmosphere.atmo4,
          ...outlineStyles,
        },
        "&:focus": {
          "& > div svg": {
            backgroundColor: theme.hv.palette.atmosphere.atmo4,
            ...outlineStyles,
          },
        },
      },
    },
  },
  /* Styles applied to the label `span` element. */
  label: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    paddingLeft: 12,
    paddingRight: 12,
    whiteSpace: "nowrap",
    maxWidth: 200,
    minHeight: 20,
  },
  square: {
    borderRadius: 2,
  },
  round: {
    borderRadius: 28,
  },

  deleteIcon: {
    WebkitTapHighlightColor: "transparent",
    height: 10,
    width: 12,
    cursor: "pointer",
    color: "white",
    "&:hover": {
      "& *": {
        backgroundColor: theme.palette.atmo3,
        ...outlineStyles,
      },
    },
  },
  focusVisible: {},
  disabled: {
    color: theme.hv.palette.atmosphere.atmo5,
    cursor: "not-allowed",
    PointerEvent: "none",
    opacity: 1,
    "&:hover": {
      color: theme.hv.palette.atmosphere.atmo5,
      border: `transparent`,
    },
    "&$focusVisible": {
      backgroundColor: theme.palette.atmo3,
      ...outlineStyles,
    },
  },
  informational: {
    backgroundColor: theme.hv.palette.semantic.sema7,
  },
  success: {
    backgroundColor: theme.hv.palette.semantic.sema8,
  },
  warning: {
    backgroundColor: theme.hv.palette.semantic.sema20,
  },
  error: {
    backgroundColor: theme.hv.palette.semantic.sema9,
  },
});
export default styles;
