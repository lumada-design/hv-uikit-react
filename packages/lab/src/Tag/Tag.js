import React from "react";
import clsx from "clsx";
import PropTypes from "prop-types";
import { withStyles, makeStyles, useTheme } from "@mui/styles";
import { Close } from "@hitachivantara/uikit-react-icons";
import { useDeprecated } from "@hitachivantara/uikit-react-core";

import styles from "./styles";

/**
 * The tag is a component used to notify the user about the labelled ,categorised, or organised item , in the app context.
 */
const useStyles = makeStyles(styles, { name: "HvTag" });

const getColor = (theme, color, defaultColor) => theme.palette[color] || color || defaultColor;
export const HvTag = (props) => {
  const {
    id,
    semantic,
    children,
    disabled,
    variant,
    color,
    shape,
    showcancelicon = false,
    ...others
  } = props;

  useDeprecated("Tag", "Please use the Tag component in Core");
  const classes = useStyles();
  const isDisabled = disabled || false;
  const Tagshape = shape || classes.square;
  const theme = useTheme();
  const inlineStyle = {};

  if (semantic) {
    inlineStyle.backgroundColor = getColor(
      theme,
      theme.palette?.[semantic],
      theme.hv.palette.accent.acce1
    );
  }

  /**
   * Renders the content for the tag elements.
   */
  return (
    <div
      id={id}
      role="button"
      className={clsx(
        classes.root,
        classes[variant],
        { ...others },
        {
          [classes.disabled]: isDisabled,
          [classes[`${shape}`]]: Tagshape,
        }
      )}
      style={inlineStyle}
      disabled={isDisabled}
      {...others}
    >
      <span className={classes.label}>{children}</span>
      {showcancelicon && (
        <Close
          className={clsx(classes.deleteIcon, {
            [classes.disabled]: isDisabled,
          })}
          color="atmo5"
        />
      )}
    </div>
  );
};

HvTag.propTypes = {
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   *   A Jss Object used to override or extend the styles applied to the tag.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,
    /**
     * Styles applied to the component label.
     */
    label: PropTypes.string,
    /**
     * Styles applied to the component when it is tag.
     */
    square: PropTypes.string,
    /**
     * Styles applied to the component when it is chip.
     */
    round: PropTypes.string,

    /**
     * Styles applied to the component when shows label.
     */
    deleteIcon: PropTypes.string,
    /**
     * Styles applied to the component when disabled.
     */
    disabled: PropTypes.string,
    /**
     * Styles applied to the button if keyboard focused.
     */
    focusVisible: PropTypes.string,
  }),
  /**
   * A String representing the background color of the tag.
   * You can use either an HEX or color name from the palette.
   */
  backgroundColor: PropTypes.string,
  /**
   * Category of tag to use
   */
  variant: PropTypes.oneOf(["informational", "success", "warning", "error", "categorical"]),
  /**
   * A String or Array of strings representing the colors to override in the icon.
   * Each element inside the array will override a diferent color.
   * You can use either an HEX or color name from the palette.
   */
  color: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  /**
   * Sets one of the standard semantic palette colors of the icon
   */
  semantic: PropTypes.oneOf([
    "sema1",
    "sema2",
    "sema3",
    "sema4",
    "sema5",
    "sema6",
    "sema7",
    "sema8",
    "sema9",
    "sema10",
    "sema11",
    "sema12",
    "sema13",
    "sema14",
    "sema15",
    "sema16",
    "sema17",
    "sema18",
    "sema19",
    "sema20",
  ]),
  shape: PropTypes.string,
  /**
   * The content of the Tag.
   */
  children: PropTypes.node.isRequired,
  /**
   * The disabled property of the Tag.
   */
  disabled: PropTypes.string,
  /**
   * The cancel icon is shown based on this property
   */
  showcancelicon: PropTypes.bool,
};

export default withStyles(styles, { name: "HvTag" })(HvTag);
