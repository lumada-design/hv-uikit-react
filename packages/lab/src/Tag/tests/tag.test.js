import React from "react";

import { render } from "@testing-library/react";

import { HvProvider } from "@hv/uikit-react-core";
import { Default } from "../stories/Tag.stories";
import "@testing-library/jest-dom/extend-expect";

describe("HvTags", () => {
  describe("sample snapshot testing", () => {
    it("Default", () => {
      const { container } = render(
        <HvProvider>
          <Default />
        </HvProvider>
      );
      expect(container).toMatchSnapshot();
    });
    it("renders a tag as expected", () => {
      const { container, getByText } = render(
        <HvProvider>
          <Default />
        </HvProvider>
      );
      const tag1 = getByText("Hello");
      const tag2 = getByText("How");
      const tag3 = getByText("Are");
      const tag4 = getByText("You");
      expect(container).toBeInTheDocument();
      expect(tag1).toBeInTheDocument();
      expect(tag2).toBeInTheDocument();
      expect(tag3).toBeInTheDocument();
      expect(tag4).toBeInTheDocument();
    });
  });
});
