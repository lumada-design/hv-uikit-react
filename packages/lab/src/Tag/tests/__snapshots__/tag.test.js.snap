// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HvTags sample snapshot testing Default 1`] = `
<div>
  <div
    category="informational"
    class="HvTag-root category"
    id="Tag1"
    role="button"
  >
    <span
      class="HvTag-label"
    >
      Hello
    </span>
  </div>
  <div
    category="informational"
    class="HvTag-root category"
    id="Tag2"
    role="button"
  >
    <span
      class="HvTag-label"
    >
      How
    </span>
    <div
      class="HvTag-deleteIcon HvIconBase-root HvIconBase-s"
      name="Close"
    >
      <svg
        focusable="false"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="color0"
          d="M8.7 8l5.3 5.3-.7.7L8 8.7 2.7 14l-.7-.7L7.3 8 2 2.7l.7-.7L8 7.3 13.3 2l.7.7z"
          fill="#999999"
        />
      </svg>
    </div>
  </div>
  <div
    category="informational"
    class="HvTag-root category HvTag-round"
    id="Tag3"
    role="button"
  >
    <span
      class="HvTag-label"
    >
      Are
    </span>
  </div>
  <div
    category="informational"
    class="HvTag-root category HvTag-round"
    id="Tag4"
    role="button"
  >
    <span
      class="HvTag-label"
    >
      You
    </span>
    <div
      class="HvTag-deleteIcon HvIconBase-root HvIconBase-s"
      name="Close"
    >
      <svg
        focusable="false"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="color0"
          d="M8.7 8l5.3 5.3-.7.7L8 8.7 2.7 14l-.7-.7L7.3 8 2 2.7l.7-.7L8 7.3 13.3 2l.7.7z"
          fill="#999999"
        />
      </svg>
    </div>
  </div>
</div>
`;
